lib/puppet/module/task.rb

```diff
@@ -157,7 +192,22 @@ def implementations
     end
 
     def files
-      implementations.map {|imp| { 'name' => imp['name'], 'path' => imp['path'] } }
+      md = metadata
+      outer_files = []
+      impl_lib_files = []
+      lib_files = []
```

Comment:
```
lib_files = md.nil? ? [] :
  self.class.find_files((md.fetch('files', []) + md['implementations']&.select { |impl| impl.key?('files') }.flat_map { |impl| impl['files'] }).uniq, @module)
```

Reply:
can start using &. and dig in puppet now?